tangelo/toolboxes/post_processing/extrapolation.py

```diff
@@ -71,41 +77,53 @@ def extrapolation(energies, coeffs, taylor_order=None):
     Eh = np.array(energies)
     coeffs = np.array(coeffs)
 
-    # Setup the linear system matrix
+    # Set up the linear system matrix
     ck = np.array([coeffs**k for k in range(1, taylor_order+1)])
     B = np.ones((n+1, n+1))
     B[n, n] = 0
     B[:n, :n] = ck.T @ ck
 
-    # Setup the free coefficients
+    # Set up the free coefficients
     b = np.zeros(n+1)
     b[n] = 1  # For the Lagrange multiplier
 
     # Solve  the DIIS equations by least squares
-    x = np.linalg.lstsq(B, b, rcond=None)[0]
-    return np.dot(Eh, x[:-1])
+    x = np.linalg.lstsq(B, b, rcond=None)[0][:-1]
+
+    if stderr is None:
+        return np.dot(Eh, x)
+    else:
+        stderr = np.array(stderr)
+        return np.dot(Eh, x), np.sqrt(np.dot(stderr**2, x**2))
 
 
-def richardson_analytical(energies, coeffs):
+def richardson_analytical(coeffs, energies, stderr=None):
     """
     Richardson extrapolation explicit result as found in
     Phys. Rev. Lett. 119, 180509 [arXiv:1612.02058]
 
     Args:
-        energies (array-like): Energy expectation values for amplified noise rates
         coeffs (array-like): Noise rate amplification factors
+        energies (array-like): Energy expectation values for amplified noise rates
+        stderr (array-like): Energy standard error estimates
 
     Returns:
         float: Extrapolated energy
+        float: Error estimation for extrapolated energy
     """
     Eh = np.array(energies)
     ck = np.array(coeffs)
-    x = np.array([np.prod(ai/(a - ai)) for i, a in enumerate(ck)
-                      for ai in [np.delete(ck, i)]])
-    return np.dot(Eh, x)
+    n = len(ck)
+    x = np.array([(-1)**(n-1) * np.prod(ai/(a - ai)) for i, a in enumerate(ck) for ai in [np.delete(ck, i)]])
+
```

Comment:
I think you can also write this as.
```python3
x = np.array([np.prod(ai/(ai - a)) for i, a in enumerate(ck) for ai in [np.delete(ck, i)]])
```

Reply:
That does seem like a simpler solution. Weird, I thought I tried that at some point. Changed